File: Python/8py 4/triple_trouble.py
"""
|--------------------------------------------------------------------------
| Triple Trouble
|--------------------------------------------------------------------------
|
"""

def triple_trouble(one, two, three):
   res = ''
   for i in range(len(one)):
      res += f'{one[i]}{two[i]}{three[i]}'
   return res


def triple_trouble(*args):
    return ''.join(sum(zip(*args), ()))


def triple_trouble(one, two, three):
   return ''.join([one[i] + two[i] + three[i] for i in range(len(one))])


def triple_trouble(*args):
   return "".join(map("".join, zip(*args)))


print(triple_trouble("aaa","bbb","ccc"), "abcabcabc")
print(triple_trouble("aaaaaa","bbbbbb","cccccc"), "abcabcabcabcabcabc")
print(triple_trouble("burn", "reds", "roll"), "brrueordlnsl")
print(triple_trouble("Bm", "aa", "tn"), "Batman")
print(triple_trouble("LLh", "euo", "xtr"), "LexLuthor")

File: JavaScript/8js 3/tripleTrouble.js
/*
|--------------------------------------------------------------------------
| Triple Trouble
|--------------------------------------------------------------------------
|
*/

function tripleTrouble(one, two, three) {
   let res = '';
   for (let i = 0; i < one.length; i++) {
      res += `${one[i]}${two[i]}${three[i]}`;
   }
   return res
}


const tripleTrouble = (a, b, c) => Array.from(a, (x, i) => x + b[i] + c[i]).join('')


function tripleTrouble(one, two, three) {
   return one.replace(/./g, (v, i) => v + two[i] + three[i])
}


function tripleTrouble(one, two, three) {
   return one.split('').reduce((acc, c, i, array) => [acc, one[i], two[i], three[i]].join(''), '')
}


function tripleTrouble(one, two, three) {
   return !arg[0][0] ? '' : arg.map(i => i[0]).join('') + tripleTrouble(...arg.map(i => i.slice(1)))
}


console.log(tripleTrouble("aaa", "bbb", "ccc"), "abcabcabc");
console.log(tripleTrouble("aaaaaa", "bbbbbb", "cccccc"), "abcabcabcabcabcabc");
console.log(tripleTrouble("burn", "reds", "roll"), "brrueordlnsl");
console.log(tripleTrouble("Sea", "urn", "pms"), "Supermans");
console.log(tripleTrouble("LLh", "euo", "xtr"), "LexLuthor");